code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm

```diff
@@ -0,0 +1,211 @@
+
+/datum/eldritch_knowledge/rust_mark/on_eldritch_blade(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(istype(target,/mob/living))
+		var/mob/living/L = target
+		L.apply_status_effect(/datum/status_effect/eldritch/rust)
+
+/datum/eldritch_knowledge/rust_blade_upgrade
+	name = "Toxic blade"
+	gain_text = "Let the blade guide you through the flesh."
+	desc = "Your blade of choice will now add toxin to enemies bloodstream."
+	cost = 2
+	next_knowledge = list(/datum/eldritch_knowledge/spell/rust_wave)
+	banned_knowledge = list(/datum/eldritch_knowledge/ash_blade_upgrade,/datum/eldritch_knowledge/flesh_blade_upgrade)
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_blade_upgrade/on_eldritch_blade(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.reagents.add_reagent(/datum/reagent/toxin,5)
+
+/datum/eldritch_knowledge/spell/rust_wave
+	name = "Wave of Rust"
+	desc = "You can now send a projectile that converts an area into rust."
+	gain_text = "Messenger's of hope fear the rustbringer!"
+	cost = 1
+	spell_to_add = /obj/effect/proc_holder/spell/targeted/projectile/dumbfire/rust_wave
+	next_knowledge = list(/datum/eldritch_knowledge/rust_final,/datum/eldritch_knowledge/spell/blood_siphon,/datum/eldritch_knowledge/summon/rusty)
+	route = "Rust"
+
+/datum/eldritch_knowledge/armor
+	name = "Armorer's ritual"
+	desc = "You can now create eldritch armor using a table and a gas mask."
+	gain_text = "For I am the heir to the throne of doom."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/flesh_ghoul)
+	required_atoms = list(/obj/structure/table,/obj/item/clothing/mask/gas)
+	result_atoms = list(/obj/item/clothing/suit/hooded/cultrobes/eldritch)
+
+/datum/eldritch_knowledge/essence
+	name = "Priest's ritual"
+	desc = "You can now transmute a tank of water into a bottle of eldritch water."
+	gain_text = "This is an old recipe, i got it from an owl."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/spell/ashen_shift)
+	required_atoms = list(/obj/structure/reagent_dispensers/watertank)
+	result_atoms = list(/obj/item/reagent_containers/glass/beaker/eldritch)
+
+/datum/eldritch_knowledge/rust_final
+	name = "Rustbringer's Oath"
+	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become more resillient overall."
+	gain_text = "Champion of rust. Corruptor of steel. Fear the dark for Rustbringer has come!"
+	cost = 3
+	required_atoms = list(/mob/living/carbon/human)
+	route = "Rust"
+	var/finished = FALSE
+
+
+/datum/eldritch_knowledge/rust_final/recipe_snowflake_check(list/atoms, loc,list/selected_atoms)
+	if(finished)
+		return FALSE
+	var/counter = 0
+	for(var/mob/living/carbon/human/H in atoms)
+		selected_atoms |= H
```

Comment:
Is the `|=` necessary over `+=` ?
Can they already be there on the list?

Reply:
yes